### ISSUE TITLE
Problems to make it working on macOs (Safari, specifically!)

### ISSUE THREAD

**Issue by mgiugliano** [DCsunset/taskwarrior-webui#20]
I just pulled the latest image, from docker-hub and launched it with

`docker run -d -p 8080:80 --name taskwarrior-webui -v $HOME/.taskrc:/.taskrc -v $HOME/mypath/to/taskfolder:/.task dcsunset/taskwarrior-webui
`
Navigating to 0.0.0.0:8080 brings up a spinning wheel but nothing happens.

Docker logs says:

`> backend@1.0.0 start
> node ./dist/app.js

Server listening on http://0.0.0.0:3000
`

I don't understand why the server (inside Docker) is listening on port 3000 instead of 80. Changing "-d -p 8080:3000" returns a web page with "Not Found" and outputs "GET / 404 1ms" on Docker logs.

With Safari developer tool, I could see the following error messages on the console:

`[Error] SyntaxError: Invalid regular expression: invalid group specifier name
	(anonymous function) (a6ee970.js:2)

[Error] ChunkLoadError: Loading chunk 2 failed.
(missing: http://0.0.0.0:8080/_nuxt/a6ee970.js)
(anonymous function) — 8310ff6.js:1:1032
component — 4448ff8.js:2:70120
v — c713f9e.js:2:83076
(anonymous function) — c713f9e.js:2:82861
r — c713f9e.js:2:72668
l — c713f9e.js:2:72872
(anonymous function) — c713f9e.js:2:72931
Promise
(anonymous function) — c713f9e.js:2:72822
(anonymous function) — 4448ff8.js:2:12923
reduce
map
B — 4448ff8.js:2:12821
D — 4448ff8.js:2:13034
v — c713f9e.js:2:83076
(anonymous function) — c713f9e.js:2:82861
r — c713f9e.js:2:72668
l — c713f9e.js:2:72872
(anonymous function) — c713f9e.js:2:72931
Promise
(anonymous function) — c713f9e.js:2:72822
(anonymous function) — 4448ff8.js:2:15245
v — c713f9e.js:2:83076
(anonymous function) — c713f9e.js:2:82861
r — c713f9e.js:2:72668
l — c713f9e.js:2:72872
(anonymous function) — c713f9e.js:2:72931
Promise
(anonymous function) — c713f9e.js:2:72822
v — c713f9e.js:2:83076
(anonymous function) — c713f9e.js:2:82861
r — c713f9e.js:2:72668
l — c713f9e.js:2:72872
promiseReactionJob

	promiseReactionJob`

Any suggestion would be appreciated. Thank you for your patience.

**Comment by DCsunset** [comment]
Hello, current nginx listens at port 80, and then it forwards the traffic to the backend. So although the backend listens at 3000, you should still bind port 80.

Could you try opening the page at `http://localhost:8080` or `http://127.0.0.1:8080`? I'm not sure whether visiting the address `0.0.0.0` would lead to undefined behaviour.

**Comment by mgiugliano** [comment]
Thank you for your time. I tried your suggestions but I still get the "spinning wheel": it spins but nothing else happens. 

The only difference between `http://127.0.0.1:8080` and `http://localhost:8080` (or `http://0.0.0.0:8080`) is that the favicon is properly rendered.

I tried ssh-ing into the running container, by `docker exec -it containername /bin/sh`. The file `/var/log/nginx/error.log` is empty and `/var/log/nginx/access.log` seems normal, logging entries like

```
172.17.0.1 - - [16/Jan/2022:14:00:20 +0000] "GET /_nuxt/04df4af.js HTTP/1.1" 304 0 "http://127.0.0.1:8080/" "Mozilla/5.0 (Macintosh; Intel Mac OS X
10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.2 Safari/605.1.15" "-"
```

No other (interesting/non-empty) file seems to be present in `/var/log` (or I do not know where to look at). I am unable to locate any possible log/error file in `/src/backend`.

**Comment by mgiugliano** [comment]
**UPDATE**

Apologies for not having tried "the obvious" earlier: using Chrome - instead of Safari - works perfectly.

How, if you are interested, may I help you making it working on Safari?

Thank you for your patience with me.

**Comment by DCsunset** [comment]
It seems that there's some dependency not compatible with Safari. I still have no idea how to fix it. Maybe I can try bumping all dependencies when I have time.

**Comment by DCsunset** [comment]
I have bumped the version of some dependencies. Maybe you can try the latest version and see if it occurs again.

**Comment by super-ben** [comment]
Unfortunately, the problem is still present even in the latest version. This is sad as my main use (90%+) would be iOS devices so at the moment I can’t actively deploy your otherwise fantastic solution.

**Comment by DCsunset** [comment]
Closed by #21.
